src/properties/propertiesService.ts

```diff
@@ -23,132 +18,71 @@ export interface PropertiesService {
   ): PropertySchemas | undefined;
 }
 
-interface PropertySchemas {
+/**
+ * Schemas describing the properties view of an editor element.
+ */
+export interface PropertySchemas {
   schema: JsonSchema;
   uiSchema?: UISchemaElement;
 }
 
-const initialSchemas = (): PropertySchemas => ({
-  schema: {
-    type: 'object',
-    properties: {},
-  },
-  uiSchema: {
-    type: 'VerticalLayout',
-    elements: [],
-  } as Layout,
-});
-
-const createControl = (controlScope: string): ControlElement => ({
-  type: 'Control',
-  scope: controlScope,
-});
-
-interface PropertiesSchemasDecorators {
+/**
+ * Decorator for the PropertySchemas of an EditorUISchemaElement.
+ */
+export interface PropertiesSchemasDecorator {
   decorate: (
     schemas: PropertySchemas,
     uiElement: EditorUISchemaElement,
     schemaElement?: SchemaElement
   ) => PropertySchemas;
 }
 
-const multilineStringOption: PropertiesSchemasDecorators = {
-  decorate: (
-    schemas: PropertySchemas,
+/**
+ * Constant that indicates that a tester is not capable of handling
+ * an EditorUISchemaElement.
+ */
+export const NOT_APPLICABLE = -1;
+
+/**
+ * Returns a PropertySchemas object for an EditorUISchemaElement. The tester will return a ranking
+ * or NOT_APPLICABLE if the provider cannot supply any schema for the given editor element. */
+export interface PropertiesSchemasProvider {
+  tester: (uiElement: EditorUISchemaElement) => number;
+  getPropertiesSchemas: (
```

Comment:
shouldn't this be named: 
```suggestion
  getPropertySchemas: (
```